test/test_store/test_store_auditable.py

```diff
@@ -1,386 +1,472 @@
+    )
+    t.commit()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_remove_add_commit(get_graph):
+    g, t = get_graph
+    t.remove((EX.s1, EX.p1, EX.o1))
+    t.add((EX.s1, EX.p1, EX.o1))
+    assert set(t) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+            (EX.s1, EX.p1, EX.o1),
+        ]
+    )
+    t.commit()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+            (EX.s1, EX.p1, EX.o1),
+        ]
+    )
+
+
+def test_add_rollback(get_graph):
+    g, t = get_graph
+    t.add((EX.s1, EX.p1, EX.o1))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_remove_rollback(get_graph):
+    g, t = get_graph
+    t.remove((EX.s0, EX.p0, EX.o0))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_multiple_remove_rollback(get_graph):
+    g, t = get_graph
+    t.remove((EX.s0, EX.p0, None))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_noop_add_rollback(get_graph):
+    g, t = get_graph
+    t.add((EX.s0, EX.p0, EX.o0))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_noop_remove_rollback(get_graph):
+    g, t = get_graph
+    t.add((EX.s0, EX.p0, EX.o0))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_add_remove_rollback(get_graph):
+    g, t = get_graph
+    t.add((EX.s1, EX.p1, EX.o1))
+    t.remove((EX.s1, EX.p1, EX.o1))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+def test_remove_add_rollback(get_graph):
+    g, t = get_graph
+    t.remove((EX.s1, EX.p1, EX.o1))
+    t.add((EX.s1, EX.p1, EX.o1))
+    t.rollback()
+    assert set(g) == set(
+        [
+            (EX.s0, EX.p0, EX.o0),
+            (EX.s0, EX.p0, EX.o0bis),
+        ]
+    )
+
+
+@pytest.fixture(
+    scope="function",
+    params=get_plugin_stores(),
+)
+def get_empty_graph(request):
+    storename = request.param
+
+    path = tempfile.mktemp()
+
+    try:
+        shutil.rmtree(path)
+    except Exception:
+        pass
+
+    g = Graph(store=storename)
+    rt = g.open(configuration=path, create=True)
+    assert rt == VALID_STORE, "The underlying store is corrupt"
+
+    t = Graph(AuditableStore(g.store), g.identifier)
+
+    yield g, t
+    g.close()
+    g.destroy(configuration=path)
```

Comment:
as far as I can tell this corresponds to

https://github.com/RDFLib/rdflib/blob/6b356a4eacde1280f78b4ed966e8a499a727189a/test/test_store/test_store_auditable.py#L225-L228

I am not sure why the version here is more complex.

Reply:
The additional complexity is in support of testing pluginstores. I've refactored the fixture contents, abstracting out the store+path setting, the store opening and the final cleanup into separate functions.

TBH, I'm not sure _how_ to best provide support for the RDFLib family ecosystem but I'm keen to do so.